File: 03a.php
<?php

$fin = fopen("03.dat", "r");
$data = fread($fin, filesize("03.dat"));
$instruction_lines = preg_split('/\n/', $data, -1, PREG_SPLIT_NO_EMPTY);
$triangles = 0;
foreach ($instruction_lines as $line) {
    $sides = explode(' ', trim(preg_replace('/\s+/', ' ', $line)));
    $is_triangle = true;
    foreach ($sides as $side) {
        $other_sides = $sides;
        array_splice($other_sides, array_search($side, $other_sides), 1);
        $side_sum = array_reduce($other_sides, fn($sum, $other_side) => intval($sum) + intval($other_side));
        if ($side_sum <= intval($side)) {
            $is_triangle = false;
            break;
        }
    }
    if ($is_triangle) {
        $triangles += 1;
    }
}
echo $triangles;

fclose($fin);

?>

File: 03a.py
#!/usr/bin/python3

import re

with open('03.dat') as f:
    data = f.read()
instruction_lines = data.split('\n')
instruction_lines.pop()
triangles = 0
for line in instruction_lines:
    sides = re.sub(r'\s+', ' ', line).strip().split(' ')
    is_triangle = True
    for side in sides:
        other_sides = sides.copy()
        other_sides.remove(side)
        side_sum = sum([int(x) for x in other_sides])
        if side_sum <= int(side):
            is_triangle = False
            break
    if is_triangle:
        triangles += 1
print(triangles)
